Code/Framework/AzCore/AzCore/Serialization/Json/JsonImporter.cpp

```diff
@@ -144,21 +160,606 @@ namespace AZ
+    }
+
+    // The algorithm below is a modified JSON Merge Patch algorithm that is for patching a specific field
+    // in the target JSON value
+    // The modifications are that any `null` entries do not revome existing fields from the target JSON
+    // The reason this is needed is because the Json Import operation should preserve existing fields
+    // in case the JSON data is used for an JSON patch or JSON merge patch operation explicitly
+    JsonSerializationResult::ResultCode JsonImportResolver::PatchField(rapidjson::Value& target,
+        rapidjson::Document::AllocatorType& allocator, const rapidjson::Value& source,
+        rapidjson::Value::StringRefType fieldNameStringRef, const rapidjson::Value& fieldValue,
+        ImportPathStack& importPathStack,
+        JsonImportSettings& settings, StackedString& element)
+    {
+        using namespace JsonSerializationResult;
+
+        using ReporterString = AZStd::fixed_string<1024>;
+
+        // Initialize the Merge Patch result to success
+        ResultCode mergePatchResult{ Tasks::Import, Outcomes::Success };
+
+        // Check if the target data already contains a field with the same name
+        rapidjson::Value* targetFieldName{};
+        rapidjson::Value* targetFieldValue{};
+        if (target.IsObject())
+        {
+            if (auto targetMemberIter = target.FindMember(fieldNameStringRef);
+                targetMemberIter != target.MemberEnd())
+            {
+                targetFieldName = &targetMemberIter->name;
+                targetFieldValue = &targetMemberIter->value;
+            }
+        }
+
+        if (fieldValue.IsObject())
+        {
+            if (targetFieldValue)
+            {
+                mergePatchResult.Combine(ResolveImportsInOrder(*targetFieldValue, allocator, fieldValue,
+                    importPathStack, settings, element));
+            }
+            else
+            {
+                // If the source is an object add an object member
+                // otherwise add an array field
+                rapidjson::Value* newFieldValue{};
+                if (source.IsObject())
+                {
+                    rapidjson::Value name;
+                    name.CopyFrom(rapidjson::Value(fieldNameStringRef), allocator, true);
+                    // Add an empty object or array field depending on the field value type
+                    target.AddMember(AZStd::move(name),
+                        rapidjson::Value{ rapidjson::kObjectType },
+                        allocator);
+                    auto newFieldMemberIter = target.FindMember(fieldNameStringRef);
+                    if (newFieldMemberIter == target.MemberEnd())
+                    {
+                        AZStd::string_view fieldName(static_cast<const char*>(fieldNameStringRef),
+                            fieldNameStringRef.length);
+                        auto reportMessage = ReporterString::format("Failed to add member %.*s to JSON while importing.",
```

Comment:
If you wanted, you could use AZ_STRING_FORMAT instead of %.*s. I tend to use the former, but I'm totally supportive of both. Just mentioning it here because it's a companion define for AZ_STRING_ARG.

Reply:
I avoid AZ_STRING_FORMAT because it is longer than "%.*s", but I used AZ_STRING_ARG because it is shorter than a `static_cast<int>(variableName.size()), variableName.data())`